librosa/core/notation.py

```diff
@@ -805,9 +824,18 @@ def key_to_notes(key: str, *, unicode: bool = True) -> List[str]:
 
         notes = notes_flat
 
+    # Apply natural signs to any note which has no other accidentals and does not appear in the scale for key.
+    if natural:
+        scale_notes = key_to_degrees(key)
```

Comment:
Since we only use this to check membership, it's much more efficient to make a python set out of the degrees:
```suggestion
        scale_notes = set(key_to_degrees(key))
```
eg:
```
In [9]: degs = librosa.key_to_degrees('Db:min')

In [10]: degs_set = set(degs)

In [11]: %timeit 5 in degs
1.27 µs ± 7.08 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [12]: %timeit 5 in degs_set
11.3 ns ± 0.0256 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
```
A silly speedup, but why not.